prow/config/config_test.go

```diff
@@ -431,6 +431,37 @@ presubmits:
 			},
 			expectError: true,
 		},
+		{
+			name:       "dup presubmits not the same branch, two files",
+			prowConfig: ``,
+			jobConfigs: []string{
+				`
+presubmits:
+  foo/bar:
+  - interval: 10m
+    agent: kubernetes
+    name: presubmit-bar
+    context: bar
+    branches:
+    - master
+    spec:
+      containers:
+      - image: alpine`,
+				`
+presubmits:
+  foo/bar:
+  - interval: 10m
+    agent: kubernetes
+    context: bar
+    branches:
+    - other
+    name: presubmit-bar
+    spec:
+      containers:
+      - image: alpine`,
+			},
+			expectError: false,
+		},
```

Comment:
also add a test case to assert same name + same branch is not allowed across main/job configs?